packages/react/src/components/login/LoginButton.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+
+import { useAuthenticatedUser, useOidcClient } from './client/hooks';
+import { useSignalTrackingWithReturnValue } from './beacon/hooks';
+import {
+  triggerForAllOidcClientStateChanges,
+  isLoggingInSignal,
+  triggerForAllOidcClientErrors,
+} from './client/signals';
+import { Button, ButtonProps } from '../button/Button';
+import { LoadingSpinner } from '../loadingSpinner';
+import { IconAlertCircleFill } from '../../icons/ui/IconAlertCircleFill';
+import styles from './LoginButton.module.css';
+
+export type LoginButtonProps = { spinnerColor?: string; errorText: string } & ButtonProps;
+export function LoginButton({
+  spinnerColor = 'var(--color-white)',
+  children,
+  errorText,
+  ...buttonProps
+}: LoginButtonProps): React.ReactElement | null {
+  const { login } = useOidcClient();
+  const user = useAuthenticatedUser();
+  const [lastSignal] = useSignalTrackingWithReturnValue(triggerForAllOidcClientStateChanges);
+  const [loginError, resetLoginError] = useSignalTrackingWithReturnValue(triggerForAllOidcClientErrors);
+  const isLoggingIn = isLoggingInSignal(lastSignal);
+  if (user) {
+    return null;
+  }
+  const isActive = isLoggingIn && !loginError;
+  // for some reason LoadingSpinner theme has no effect
+  const iconLeft = isActive ? (
+    <LoadingSpinner small style={{ '--spinner-color': spinnerColor } as React.HTMLProps<HTMLDivElement>['style']} />
+  ) : undefined;
+  const combinedButtonProps = {
+    ...buttonProps,
+    iconLeft,
+    disabled: isActive,
+    onClick: () => {
+      resetLoginError();
+      login();
+    },
+  };
+  return (
+    <>
+      <div className={styles.container}>
+        <Button {...combinedButtonProps}>{children}</Button>
+        {loginError && (
+          <div className={styles.loginError}>
+            <IconAlertCircleFill className={styles.icon} />
+            <span className={styles.text}>{errorText}</span>
+          </div>
+        )}
+      </div>
+    </>
```

Comment:
Won't probably add the new component to this PR.

Reply:
I agree, separate PR for
- action bar login button using this login system ( I don't know if it would be react-way to inherit the loginbutton, but you get the idea, passing the login specific features to the different looking button)
- documentation how to set it up in any different button (maybe in the customisation-tab?)